aiocache/decorators.py

```diff
@@ -72,18 +67,21 @@ def __call__(self, f):
         return wrapper
 
     async def decorator(self, f, *args, **kwargs):
-        self._conn = self.cache.get_connection()
-
-        async with self._conn:
-            key = self.get_cache_key(f, args, kwargs)
+        key = self.get_cache_key(f, args, kwargs)
 
-            value = await self.get_from_cache(key)
+        try:
```

Comment:
Please keep the `get_from_cache` and `set_in_cache` functions. In the end both `cached` and `cached_stampede` will use the same code now :)